iam_pipes.tf

```diff
@@ -0,0 +1,396 @@
+        "ec2:DeleteNetworkInterface",
+        "ec2:DescribeNetworkInterfaces",
+        "ec2:DescribeSecurityGroups",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeVpcs",
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+
+    lambda = {
+      actions = [
+        "lambda:InvokeFunction"
+      ]
+    }
+
+    step_functions = {
+      actions = [
+        "states:StartExecution"
+      ]
+    }
+
+    api_gateway = {
+      actions = [
+        "execute-api:Invoke"
+      ]
+    }
+
+    api_destination = {
+      actions = [
+        "events:InvokeApiDestination"
+      ]
+    }
+
+    batch = {
+      actions = [
+        "batch:SubmitJob"
+      ]
+    }
+
+    logs = {
+      actions = [
+        "logs:DescribeLogGroups",
+        "logs:DescribeLogStreams",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+
+    ecs = {
+      actions = [
+        "ecs:RunTask"
+      ]
+    }
+
+    ecs_iam_passrole = {
+      actions = [
+        "iam:PassRole"
+      ]
+    }
+
+    eventbridge = {
+      actions = [
+        "events:PutEvents"
+      ]
+    }
+
+    firehose = {
+      actions = [
+        "firehose:PutRecord"
+      ]
+    }
+
+    inspector = {
+      actions = [
+        "inspector:CreateAssessmentTemplate" # ???
+      ]
+    }
+
+    redshift = {
+      actions = [
+        "redshift-data:ExecuteStatement" # ???
+      ]
+    }
+
+    sagemaker = {
+      actions = [
+        "sagemaker:CreatePipeline" # ???
+      ]
+    }
+
+    sns = {
+      actions = [
+        "sns:Publish"
+      ]
+    }
+  }
+}
+
+#################################
+# IAM role for EventBridge Pipes
+#################################
+
+data "aws_iam_policy_document" "assume_role_pipe" {
+  for_each = local.role_for_pipes
+
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["pipes.amazonaws.com"]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [data.aws_caller_identity.current.account_id]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceArn"
+      values   = ["arn:aws:pipes:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:pipe/${each.value.Name}"]
+    }
+  }
+}
+
+resource "aws_iam_role" "eventbridge_pipe" {
+  for_each = local.role_for_pipes
+
+  name                  = each.value.role_name
+  description           = try(each.value.role_description, null)
+  path                  = try(each.value.role_path, null)
+  force_detach_policies = try(each.value.role_force_detach_policies, null)
+  permissions_boundary  = try(each.value.role_permissions_boundary, null)
+  assume_role_policy    = data.aws_iam_policy_document.assume_role_pipe[each.key].json
+
+  tags = merge({ Name = each.value.role_name }, var.tags)
```

Comment:
just curious - do we still need to follow the pattern of merging in `{ Name = "<xxx>"}` into tags since nearly all resources now support a name attribute field (there may be a few left that still inherit the name from the `Name` tag and those are valid use cases)

Reply:
Maybe not, added support for `role_tags` per role to be able to override `Name`.

There are some confusions with `Name` vs `name` in this module, but I will resolve it in the future release (breaking change, probably).